### DISCUSSION TITLE
keycloak-admin-client CORS Setting in Angular App

### DISCUSSION THREAD

**Discussion by MrTob** [keycloak/keycloak#26372] [Q&A]
Hello,
I would like to add users to my Keycloak via the admin interface, so I have added the 'npm' package to my Angular application.
If I now want to add a user I get the following error.
`Access to fetch at 'http://localhost:8080/realms/master/protocol/openid-connect/token' from origin 'http://localhost:4200' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`
I think I have to specify the settings somewhere when creating the `KcAdminClient.` Can anyone help me?

**Comment by jonkoops** [comment]
> I would like to add users to my Keycloak via the admin interface, so I have added the 'npm' package to my Angular application.

If you want to access the Admin Console you don't have to add the Admin Client library, that is to make calls to the Admin REST API. If you want to manage users from the Admin Console you can redirect to there instead.

**Reply by MrTob** [reply]
But I want to create new users via the REST interface because I have my own registration flow in my Angular app.

**Reply by jonkoops** [reply]
Ok, that wasn't really clear from your original statement. If you want to make calls to the Admin REST API you'll likely have to configure your client first, see this section on [Keycloak server configuration](https://www.keycloak.org/docs/latest/securing_apps/#keycloak-server-configuration) of the JS adapter documentation.

**Reply by MrTob** [reply]
i tried to [override ](https://www.npmjs.com/package/@keycloak/keycloak-admin-client?activeTab=readme)the default Config
```
const kcAdminClient = new KcAdminClient({
          baseUrl: 'http://localhost:8080',
          realmName: 'master',
});
```
but when i call the `auth()` function - a CORS error in the Console will appear.
![image](https://github.com/keycloak/keycloak/assets/30464573/077bb34d-1444-4517-bfe7-d5526b73da2f)
I have not found anything in the documentation where I can edit the headers, is this the solution?

**Reply by jonkoops** [reply]
Make sure that your Keycloak server is properly configured, like the docs I sent state:

> You also need to configure Valid Redirect URIs and Web Origins.
